src/server/Server.java

```diff
@@ -9,92 +15,72 @@
-        final PrintWriter pw = new PrintWriter(sw, true);
-        throwable.printStackTrace(pw);
-        return sw.getBuffer().toString();
-    }
-
-    public void setSOTimeout(int in) {
-        SOTimeout = in;
-    }
-
-    public void setBufferSize(int in) {
-        BufferSize = in;
-    }
-
-    public void setMaximumConcurrentRequests(int in) {
-        MaxConcurrentRequests = in;
-        backlog = MaxConcurrentRequests * 5;
-    }
+    public final int port;
+    public final int maxRequestSizeBytes;
+    public final int maxConcurrentRequests;
+    public final int backlog;
+    public final boolean useGzip;
+    public final String rootDir;
+    public final String mimeFile;
 
-    public void setGZip(boolean in) {
-        GZip = in;
-    }
+    private final SparkDB MIME = new SparkDB();
+    private final Function<HttpRequest, HttpResponse> handler;
 
-    public void setMaximumRequestSizeInKB(int in) {
-        MaxRequestSizeKB = in;
-    }
 
-    public void setBacklog(int in) {
-        backlog = in;
+    public Server(Function<HttpRequest, HttpResponse> handler) {
+        this(8080, false, handler);
     }
 
-    public void setMIMEFile(String in) throws Exception {
-        MIMEFile = in;
-        loadMIME();
+    public Server(int port, boolean useGzip, Function<HttpRequest, HttpResponse> handler) {
+        this(port, 100000, 4096, 4096, useGzip, handler);
     }
 
-    public void setWWWDirectory(String in) {
-        WWWDir = in;
+    public Server(int port, int maxConcurrentRequests, int maxRequestSizeBytes, int maxResponseSizeBytes, boolean useGzip, Function<HttpRequest, HttpResponse> handler) {
+        this(port, maxConcurrentRequests, maxRequestSizeBytes, useGzip, "www", "etc/MIME.db", handler);
     }
 
-    public void setPort(int in) {
-        Port = in;
+    public Server(int port, int maxConcurrentRequests, int maxRequestSizeBytes, boolean useGzip, String rootDir, String mimeFile, Function<HttpRequest, HttpResponse> handler) {
+        this.port = port;
+        this.maxConcurrentRequests = maxConcurrentRequests;
+        this.backlog = maxConcurrentRequests * 5;
+        this.maxRequestSizeBytes = maxRequestSizeBytes;
+        this.useGzip = useGzip;
+        this.rootDir = rootDir;
+        this.mimeFile = mimeFile;
+        this.handler = handler;
     }
 
-    public void addCustomHeader(Entry<String, String> in) {
-        CustomHeaders.put(in.getKey(), in.getValue());
+    private static String getStackTrace(final Throwable throwable) {
+        final StringWriter sw = new StringWriter();
+        final PrintWriter pw = new PrintWriter(sw, true);
+        throwable.printStackTrace(pw);
+        return sw.getBuffer().toString();
     }
 
     private void mainLoop(ServerSocket serverSocket, ThreadPoolExecutor poolExecutor) {
         while (true) {
             try {
                 Socket s = serverSocket.accept();
-                s.setKeepAlive(false);
+                s.setKeepAlive(true);
```

Comment:
Is Slowloris mitigated? Also, the user should customize it.

Reply:
DOS attacks just need powerful hardware XD